kurl_proxy/assets/tls.js

```diff
@@ -1,145 +1,190 @@
-if (document.readyState !== "loading") {
-  ready();
-} else {
-  document.addEventListener('DOMContentLoaded', ready);
-}
-
-function ready() {
-  var form = document.getElementById("upload-form");
-  if (form) {
-    form.addEventListener("submit", uploadAndWait);
+function startTLS() {
+  let useSelfSigned = true;
+
+  let selfSignedLabels;
+  let customCertLabels;
+  if (document.readyState !== "loading") {
+    ready();
+  } else {
+    document.addEventListener('DOMContentLoaded', ready);
   }
 
-  var skip = document.getElementById("skip-button");
-  if (skip) {
-    skip.addEventListener("click", skipAndWait);
-  }
-}
+  function ready() {
+    selfSignedLabels = document.getElementsByClassName("self-signed-visible");
+    customCertLabels = document.getElementsByClassName("custom-cert-visible");
 
-function uploadAndWait(e) {
-  e.preventDefault();
+    function handleSubmit(e) {
 
-  var formData = new FormData();
+      if (useSelfSigned) {
+        skipAndWait(e);
+        return;
+      }
 
-  var certInput = document.getElementById("cert");
-  var keyInput = document.getElementById("key");
-  var hostnameInput = document.getElementById("hostname");
+      uploadAndWait(e);
```

Comment:
now that there is only one button we need to fork the behavior based on the radio button value